trio/_highlevel_generic.py

```diff
@@ -35,7 +37,9 @@ async def aclose_forcefully(resource):
 
 
 @attr.s(eq=False, hash=False)
-class StapledStream(HalfCloseableStream):
+class StapledStream(
```

Comment:
I think StapledStream is safe enough to allow subclassing of, since it has no private attributes.

Reply:
Yeah, the cost of supporting subclasses would be relatively low compared to other things. But that's only the "cost" side of the cost/benefit analysis :-). Is there some reason `StapledStream` subclassing is particularly valuable? I'd rather keep things simple until there's a good reason not too.

Reply:
I don't have an immediate use case and am happy to leave it non-subclassable for consistency.

Reply:
:+1: We can always change it later.